lib/ES/Util.pm

```diff
@@ -45,42 +45,88 @@ sub build_chunked {
     my $resources = $opts{resource}      || [];
     my $noindex   = $opts{noindex}       || '';
     my $page_header = custom_header($index) || $opts{page_header} || '';
+    my $asciidoctor = $opts{asciidoctor} || 0;
+
     $dest->rmtree;
     $dest->mkpath;
 
+    my %xsltopts = (
+            "toc.max.depth"            => 5,
+            "toc.section.depth"        => $chunk,
+            "chunk.section.depth"      => $chunk,
+            "local.book.version"       => $version,
+            "local.book.multi_version" => $multi,
+            "local.page.header"        => $page_header,
+            "local.book.section.title" => "Learn/Docs/$section",
+            "local.book.subject"       => $subject,
+            "local.noindex"            => $noindex,
+    );
+
     my ( $output, $died );
-    eval {
-        $output = run(
-            'a2x', '-v',    #'--keep',
-            '--icons',
-            ( map { ( '--resource' => $_ ) } @$resources ),
-            '-d' => 'book',
-            '-f' => 'chunked',
-            '-a' => 'showcomments=1',
-            '-a' => "lang=$lang",
-            '-a' => 'base_edit_url=' . $edit_url,
-            '-a' => 'root_dir=' . $root_dir,
-            $private ? ( '-a' => 'edit_url!' ) : (),
-            '--xsl-file'      => 'resources/website_chunked.xsl',
-            '--asciidoc-opts' => '-fresources/es-asciidoc.conf',
-            '--destination-dir=' . $dest,
-            ( $lenient ? '-L' : () ),
-            docinfo($index),
-            xsltopts(
-                "toc.max.depth"            => 5,
-                "toc.section.depth"        => $chunk,
-                "chunk.section.depth"      => $chunk,
-                "local.book.version"       => $version,
-                "local.book.multi_version" => $multi,
-                "local.page.header"        => $page_header,
-                "local.book.section.title" => "Learn/Docs/$section",
-                "local.book.subject"       => $subject,
-                "local.noindex"            => $noindex
-            ),
-            $index
-        );
-        1;
-    } or do { $output = $@; $died = 1; };
+    if ( $asciidoctor ) {
+        # TODO icons
+        my $chunks_path = dir("$dest/.chunked");
+        $chunks_path->mkpath;
+        # Emulate asciidoc_dir because we use it to find shared asciidoc files
+        # but asciidoctor doesn't support it.
+        my $asciidoc_dir = dir('resources/asciidoc-8.6.8/')->absolute;
```

Comment:
I'm not particularly concerned that we'll upgrade and lose it. To be honest I don't think we'll upgrade asciidoc at this point. I can't comment on whether or not the path should have a version number in it because it had one when I got here.